### PR TITLE
feat(cli): TTY-detect at REPL + engineer --json (issue #62 items 1, 2, 6)

### PR DESCRIPTION
**Pull request by kalil0321**
**Stacked on top of #61** (`feat/agent-friendly-cli`). Once #61 lands, the base will rebase to `main`.

## Summary

Knocks out the first two blocker items from #62 (agent friendliness) plus a partial fix for item #6.

### Item #1 — TTY-detect at REPL entry

Without a TTY and no subcommand, `reverse-api-engineer` (bare) would block on `prompt_toolkit` forever — a footgun for CI invocations or wrappers that forget the subcommand. Now exits 2 with `--help` printed to stderr.

\`\`\`bash
$ reverse-api-engineer < /dev/null; echo \$?
... (--help output) ...
error: no subcommand given and stdin is not a TTY; the interactive REPL requires a terminal.
2
\`\`\`

### Item #2 — `engineer --json` / `--no-interactive`

Mirrors the agent-mode JSON contract (#61). Stable schema:
\`\`\`json
{
  \"schema_version\": 1,
  \"status\": \"ok\" | \"error\",
  \"run_id\": \"<id>\",
  \"prompt\": \"...\" | null,
  \"fresh\": false,
  \"script_path\": \"/abs/api_client.py\" | null,
  \"usage\": { \"input_tokens\": ..., \"output_tokens\": ..., \"total_cost\": ... },
  \"error\": \"<message>\" | null
}
\`\`\`

Now an agent can chain the full pipeline end-to-end:
\`\`\`
agent --json  →  run_id  →  engineer <run_id> --json  →  run <run_id> --no-interactive
\`\`\`

### Item #6 (partial) — root `--help` advertises scripted features

Added a one-liner in the root command docstring so an agent inspecting `--help` discovers that subcommands accept `--json` / `--no-interactive`.

## Still open from #62
- #3 normalize \`usage\` to a stable subset (Med, ~15 lines)
- #4 \`error_kind\` enum in error payloads (Med, ~10 lines)
- #5 \`agent --dry-run\` (Med, ~50 lines)
- #8 \`run --json\` wrapped (Low, ~25 lines)
- #9 \`--json-schema-version\` (Low, ~5 lines)

## Test plan
- [x] \`uv run pytest tests/test_cli_followups.py\` (14/14 — TTY-detect via real subprocess, \`_build_engineer_payload\` edge cases, engineer click wiring with success/not-found/KeyboardInterrupt/Exception, --prompt vs --fresh threading)
- [x] \`uv run pytest\` full suite — 673 passing; 5 pre-existing failures all reproduce on the base branch
- [x] Manual smoke: \`reverse-api-engineer < /dev/null\` exits 2; \`engineer doesnotexist --json\` returns full schema with status=error
- [ ] End-to-end smoke of an actual capture+engineer chain before merge

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Adds TTY detection to prevent REPL hangs, adds `engineer --json`/`--no-interactive` with a stable JSON output, and adds `--headless` to `agent` for CI/VPS use; non-interactive runs now skip follow-up prompts. Addresses issue #62 items 1, 2, and partially 6; root `--help` now mentions scripted flags, and `manual` is explicitly human-only.

- New Features
  - `engineer --json` emits one JSON payload (logs to stderr) with: schema_version, status, run_id, prompt, fresh, script_path, usage, error; exits 0 on ok, 1 on error. `--no-interactive` implies non-interactive.
  - `agent --headless`: For MCP, `chrome-mcp` headless drops `--autoConnect` and adds `--headless`; `playwright` adds `--headless`. `manual` stays human-only and does not accept `--headless` (help text points agents to `agent --headless`).

- Bug Fixes
  - No subcommand + non-TTY prints `--help` to stderr and exits 2 instead of hanging.
  - In `--json` or `--no-interactive` modes, follow-up prompts are fully suppressed: `interactive=False` is threaded through `engineer`, `agent`, `run_reverse_engineering`, and into `BaseEngineer` (whose `_prompt_follow_up()` now returns immediately).

<sup>Written for commit 0e5ef06db77aa36d86245173948624202694a214. Summary will update on new commits.</sup>

<h3>Greptile Summary</h3>

This PR adds two agent-friendliness improvements to the CLI: a TTY guard that prevents `prompt_toolkit` from blocking indefinitely when the tool is invoked without a subcommand in a non-TTY environment (exits 2 with help on stderr), and a `--json` / `--no-interactive` flag pair on the `engineer` subcommand that mirrors the existing `agent --json` contract.

- **TTY detection** (`main`): checks `sys.stdin.isatty()` before entering `repl_loop`; prints help to stderr and calls `sys.exit(2)` on failure — tested end-to-end with a real subprocess.
- **`engineer --json`** (`engineer` command + `_build_engineer_payload`): wraps `run_engineer` in `_quiet_consoles_for_json`, catches `KeyboardInterrupt` and generic exceptions, and emits a stable JSON schema on stdout with exit code 0/1; two minor style inconsistencies with the sibling `_build_agent_payload` function noted.
- **Root `--help` update**: adds a one-liner advertising `--json` / `--no-interactive` support to the root command docstring.

<h3>Confidence Score: 4/5</h3>

Safe to merge; the new code paths are well-tested and follow an established pattern already used by the agent command.

The TTY guard and engineer JSON path are straightforward, well-covered by 14 new tests (subprocess + unit + Click wiring), and mirror a pattern already proven by the agent command. Two cosmetic inconsistencies between _build_engineer_payload and _build_agent_payload were flagged but neither affects correctness.

No files require special attention; the only items worth a second glance are the redundant isinstance guards in _build_engineer_payload.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/reverse_api/cli.py | Adds `_build_engineer_payload`, TTY-guard in `main`, and `--json`/`--no-interactive` flags to the `engineer` command. Logic is sound and mirrors the existing agent JSON path; two minor style inconsistencies with `_build_agent_payload` noted. |
| tests/test_cli_followups.py | New test file covering TTY detection (via real subprocess), payload-builder edge cases, Click command wiring, and `--prompt`/`--fresh` threading. Good coverage of success, not-found, KeyboardInterrupt, and exception paths. |

</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 2 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 2
src/reverse_api/cli.py:138-150
After line 138 reassigns `result` to `{}` when it is `None`, `result` is always a `dict`. The `isinstance(result, dict)` guards on the next three lines are therefore always `True` and the branches they guard against can never fire. `_build_agent_payload` does the same normalization without any `isinstance` guards, and the inconsistency may confuse future readers. Simpler to mirror the agent function's style.

```suggestion
    result = result if result is not None else {}
    inner_error = result.get("error")
    final_error = error or inner_error or (None if result else "engineering produced no result")
    return {
        "schema_version": AGENT_JSON_SCHEMA_VERSION,
        "status": "error" if final_error else "ok",
        "run_id": run_id,
        "prompt": prompt,
        "fresh": fresh,
        "script_path": result.get("script_path"),
        "usage": result.get("usage") or {},
        "error": final_error,
    }
```

### Issue 2 of 2
src/reverse_api/cli.py:151-152
PEP 8 requires two blank lines between top-level definitions. There is only one blank line between the closing brace of `_build_engineer_payload` and the `# Mode definitions` comment, while everywhere else in this file top-level blocks are separated by two blank lines.

`````

</details>

<sub>Reviews (1): Last reviewed commit: ["feat(cli): TTY-detect at REPL entry + en..."](https://github.com/kalil0321/reverse-api-engineer/commit/d2c4e49b8692a1f81cd1b77f8cfa4dfa9e27017b) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=30900547)</sub>

### CODE REVIEW

File: src/reverse_api/cli.py [hunk 1]
```diff
@@ -1389,20 +1461,59 @@ def run_auto_capture(prompt=None, url=None, model=None, output_dir=None, agent_p
...
+    with _quiet_consoles_for_json() as real_stdout:
+        try:
+            result = run_engineer(
+                run_id,
+                prompt=main_prompt,
+                additional_instructions=additional,
+                model=model,
+                output_dir=output_dir,
+                is_fresh=fresh,
+            )
```
[the remark this answers is not in the document]
Comment (kalil0321) [reply src/reverse_api/cli.py:1558]: **Ack** — fixed in ea9ea77. Confirmed by reading ClaudeEngineer.analyze_and_generate (engineer.py:172-176): the conversation loop calls `_prompt_follow_up()` after the first generation, which awaits `input("  > ")` via a thread executor. Same bug exists in `auto_engineer.py:170` so `agent --json` had the same latent issue.

Fix:
- Added `interactive: bool = True` to BaseEngineer; `_prompt_follow_up` short-circuits to `None` when `interactive=False` (still flushes sync so partial output reaches disk)
- Threaded `interactive` through `run_reverse_engineering`, all three SDK constructors, `run_engineer`, `run_auto_capture`, `run_agent_capture` and the click commands
- `engineer` and `agent` derive `interactive = not (as_json or no_interactive)` and pass it down
- Added 6 regression tests in `test_cli_followups.py`, including one that patches `builtins.input` to raise so it'd fail loudly if the short-circuit ever regressed
